pyat/at/lattice/lattice_object.py

```diff
@@ -312,21 +301,125 @@ def _addition_filter(self, elems: Iterable[Element]):
+            :py:obj:`Element` as input argument.
+
+            Equivalents syntaxes:
+            >>>ring.append(elem)
+            >>>ring += [elem]
+
+            Parameters:
+                elem (Element): AT element to be appended to the lattice
+
+            Keyword Arguments:
+                copy(bool): Default :py:obj:`True`. If :py:obj:`True`
+                            a deep copy of elem is used
+        """
+        self.extend([elem], copy=copy)
+
+    def repeat(self, n: int, copy=True):
+        r"""This method allows to repeat the lattice `n` times.
+            If `n` does not divide `ring.periodicity`, the new ring
+            periodicity is set to 1, otherwise  it is et to
+            `ring.periodicity /= n`.
+
+            Equivalents syntaxes:
+            >>>newring = ring.repeat(n)
+            >>>newring = ring * n
+
+            Parameters:
+                n (int): number of repetition
+
+            Keyword Arguments:
+                copy(bool): Default :py:obj:`True`. If :py:obj:`True`
+                            deepcopies of the lattice are used for the
+                            repetition
+
+            Returns:
+                newring (Lattice): the new repeated lattice
+        """
+        periodicity = self.periodicity
+        if n != 0 and periodicity > 1:
+            nbp = periodicity / n
+            periodicity = int(round(nbp))
+            if abs(periodicity - nbp) > _TWO_PI_ERROR:
+                warn(AtWarning('Non-integer number of cells: {}/{}. Periodi'
+                               'city set to 1'.format(self.periodicity, n)))
+                periodicity = 1
+        elems = itertools.repeat(self, max(n-1, 0))
+        lattice = self.concatenate(*elems, copy=copy)
+        return Lattice(lattice, periodicity=periodicity)
+
+    def concatenate(self, *lattices: Tuple[Iterable[Element], ...],
+                    copy=False, inplace=False):
+        """Concatenate several `Iterable[Element]` with the lattice
+
+        Equivalents syntaxes:
+        >>>newring = ring.concatenate(r1, r2, r3)
+        >>>newring = ring + r1 + r2 + r3
+
+        Parameters:
+            lattices: :py:obj:`Iterables[Element]` to be concatenanted
+                      to the Lattice
+
+        Keyword Arguments:
+            copy(bool): Default :py:obj:`False`. If :py:obj:`True`
+                        deepcopies of the elements of lattices are used
+            inplace(bool): Default :py:obj:`False`. If :py:obj:`True`
+                           the lattice is modified in place.
+                           Oterwise a new Lattice object is returned
+
+        Returns:
+            newring(Lattice): concatenated Lattice
+        """
+        if inplace:
+            lattice = self
+        else:
+            lattice = Lattice(self)
+        for lat in numpy.atleast_2d(lattices):
+            lattice.extend(lat, copy=copy)
+        return lattice
```

Comment:
For safety, it should return lattice only if `inplace=False`:
```
return None if inplace else lattice
```

Reply:
I have to return `lattice` otherwise the operator `+=` does not work

Reply:
I take it back I have modified `__iadd__` and it is fine (I also mixed up `inplace` and `copy` that have inverse logic...)